### PR TITLE
[FSDP] Update `ShardingStrategy` and `_free_full_params()` docs

### PR DESCRIPTION
**Pull request by awgu**
1. I messed up the comment for the post-backward `_free_full_params()` in https://github.com/pytorch/pytorch/pull/75901.
This removes the comment, which is not necessary, and instead adds an explanation in the `SHARD_GRAD_OP` comment itself.
2. This updates the overall `ShardingStrategy` documentation after the observation that `SHARD_GRAD_OP` did not specify that parameters are still sharded outside of computation.

### CODE REVIEW

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -126,7 +126,8 @@ class ShardingStrategy(Enum):
                    before forward computation and keeps the full parameters in
                    GPU memory until backward computation is done. It inserts reduce_scater
                    after backward computation for synchronizing and sharding gradients.
-                   Sharded optimizer states are updated locally.
+                   Sharded optimizer states are updated locally. Inside ``no_sync()``,
```
Comment (rohan-varma) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:129]: I wonder if the beginning of this doc should be updated too, as parameters are still technically sharded, they are just in their full states after forward and through the backward

**Review by rohan-varma — COMMENTED** [review]
Thanks for the fix!

File: torch/distributed/fsdp/fully_sharded_data_parallel.py (hunk 1 above)
Comment (awgu) [reply torch/distributed/fsdp/fully_sharded_data_parallel.py:129]: Good point. I will update this doc (soon).

**Review by rohan-varma — APPROVED** [review]

**Comment by awgu** [comment]
@rohan-varma Re-requesting review because I updated the `ShardingStrategy` documentation. I have changed the PR title to reflect this accordingly.

**Review by rohan-varma — APPROVED** [review]
LGTM

**Review by zhaojuanmao — APPROVED** [review]
looks great, thank you!

**Pull request closed without merging**